frontend/public/components/secrets/create-secret.tsx

```diff
@@ -0,0 +1,155 @@
+      secret: secret,
+      inProgress: false,
+      type: secret.type,
+      stringData: _.mapValues(secret.data, (v) => window.atob(v)),
+    };
+    this.onNameChanged = this.onNameChanged.bind(this);
+    this.save = this.save.bind(this);
+  }
+  onDataChanged (secretsData) {
+    let stringData = {...this.state.stringData};
+    stringData = secretsData;
+    this.setState({stringData});
+  }
+  onNameChanged (event) {
+    let secret = {...this.state.secret};
+    secret.metadata.name = event.target.value;
+    this.setState({secret});
+  }
+  save (e) {
+    e.preventDefault();
+    const { kind, metadata } = this.state.secret;
+    this.setState({ inProgress: true });
+    const newSecret = _.assign({}, this.state.secret, {stringData: this.state.stringData});
+    const ko = kindObj(kind);
+    (this.props.isCreate
+      ? k8sCreate(ko, newSecret)
+      : k8sUpdate(ko, newSecret, metadata.namespace, newSecret.metadata.name)
+    ).then(
+      () => {
+        this.setState({inProgress: false});
+        if (metadata.namespace) {
+          this.props.setActiveNamespace(metadata.namespace);
+        }
+        history.push(formatNamespacedRouteForResource('secrets'));
+      },
+      err => this.setState({error: err.message, inProgress: false})
+    );
+  }
+  render () {
+    const title = `${this.props.titleVerb} ${_.upperFirst(this.state.secretType)} Secret`;
+    const { saveButtonText } = this.props;
+
+    let explanation = 'Webhook secret allow you to authenticate a webhook trigger.';
+    let subform = <WebHookSecretSubform onChange={this.onDataChanged.bind(this)} stringData={this.state.stringData} />;
+
+    return <div className="co-m-pane__body">
+      <Helmet>
+        <title>{title}</title>
+      </Helmet>
+      <form className="co-m-pane__body-group" onSubmit={this.save}>
+        <h1 className="co-m-pane__heading">{title}</h1>
+        <p className="co-m-pane__explanation">{explanation}</p>
+
+        <fieldset disabled={!this.props.isCreate}>
+          <div className="form-group">
+            <label className="control-label">Secret Name</label>
+            <div>
+              <input className="form-control" type="text" onChange={this.onNameChanged} value={this.state.secret.metadata.name} required id="test--subject-name" />
+              <p className="help-block">Unique name of the new secret.</p>
+            </div>
+          </div>
+        </fieldset>
+        {subform}
+        <ButtonBar errorMessage={this.state.error} inProgress={this.state.inProgress} >
+          <button type="submit" className="btn btn-primary" id="create-secret">{saveButtonText || 'Create'}</button>
+          <Link to={formatNamespacedRouteForResource('secrets')} className="btn btn-default">Cancel</Link>
+        </ButtonBar>
+      </form>
+    </div>;
+  }
+}
+
+const BaseEditSecret = connect(null, {setActiveNamespace: UIActions.setActiveNamespace})(
+  (props: BaseEditSecretProps_) => <BaseEditSecret_ {...props} />
+);
```

Comment:
Can't this just be...

```js
const BaseEditSecret = connect(null, {setActiveNamespace: UIActions.setActiveNamespace})(BaseEditSecret_);
```

Reply:
No the `props` need to be passed explicitly from parent to child otherwise, yarn will scream that some properties does not exist:
> Property 'fixed' does not exist on type 'IntrinsicAttributes & IntrinsicClassAttributes<Component<Pick<any, never>, ComponentState, never>...'.

Reply:
Then I think we must have declared the types incorrectly somehow. We should figure out what's wrong.

Reply:
The issue is that the redux part of the code is in jsx an therefor is not typed, eg. [ui-actions.js](https://github.com/openshift/console/blob/master/frontend/public/ui/ui-actions.js). I think until we type the redux part of the then codebase, which means rewriting it to TS, until then we need this as a workaround.